### PR TITLE
Use .find().map() instead of .filter_map().next()

### PR DESCRIPTION
**Pull request by fstr**
Patch for issue #9403

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9436)

### CODE REVIEW

File: components/script/dom/urlsearchparams.rs [hunk 1]
```diff
@@ -78,13 +78,7 @@ impl URLSearchParamsMethods for URLSearchParams {
...
     fn Get(&self, name: USVString) -> Option<USVString> {
         let list = self.list.borrow();
-        list.iter().filter_map(|&(ref k, ref v)| {
-            if k == &name.0 {
-                Some(USVString(v.clone()))
-            } else {
-                None
-            }
-        }).next()
+        list.iter().find(|&&(ref k, _)| k == &name.0).map(|&(_, ref v)| USVString(v.clone()))
```
Comment (KiChjang) [inline components/script/dom/urlsearchparams.rs]: Could this have one less & in the `.find()` method?

Comment (KiChjang) [inline components/script/dom/urlsearchparams.rs]: Actually, is it possible to not have `|&(ref k, _)|` at all and simply have `|ref kv| k.0 == name.0` instead?

Comment (nox) [inline components/script/dom/urlsearchparams.rs]: Or just `|&kv| kv.0 == name.0` and `|&kv| USVString(kv.1.clone())`.

**Comment by frewsxcv** [comment]
In its current state, I find the previous code much more readable than with these changes

**Comment by frewsxcv** [comment]
...though I think that can be improved if it was broken up into one line per each method call

**Comment by KiChjang** [comment]
Agreed, let's break it down into two lines.

**Comment by fstr** [comment]
I tried to implement all the suggestions.

It's also possible to not work with `&` and `ref`. Is there any disadvantage?

```
list.iter().find(|kv| kv.0 == name.0)
    .map(|kv| USVString(kv.1.clone()))
```

**Comment by KiChjang** [comment]
r=me post-nit, it's definitely fine to just use `&kv`.

@bors-servo delegate+

**Comment by fstr** [comment]
```
find(|&kv| ...).map(|&kv| ...)
```

```
/Users/florian/servo/components/script/dom/urlsearchparams.rs:82:19: 82:22 error: cannot move out of borrowed content [E0507]
/Users/florian/servo/components/script/dom/urlsearchparams.rs:82             .map(|&kv| USVString(kv.1.clone()))
                                                                                   ^~~
/Users/florian/servo/components/script/dom/urlsearchparams.rs:82:20: 82:22 note: attempting to move value to here
/Users/florian/servo/components/script/dom/urlsearchparams.rs:82             .map(|&kv| USVString(kv.1.clone()))
                                                                                    ^~
/Users/florian/servo/components/script/dom/urlsearchparams.rs:82:20: 82:22 help: to prevent the move, use `ref kv` or `ref mut kv` to capture value by reference
```

So I leave it as it is.

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Comment by jdm** [comment]
@bors-servo: retry
* travis infrastructure problem?

**Pull request merged**